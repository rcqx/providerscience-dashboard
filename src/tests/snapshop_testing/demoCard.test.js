import renderer from 'react-test-renderer';
import DemoCard from '../../components/bookedDemos/demoCard';

// Snapshot test for DemoCard component
it('renders correctly', () => {
  const mockCard = {
    Name: 'John Doe',
    number: 1,
    profilePicture: 'aString',
    demos: 5,
    percentage: 50,
    madeCalls: 1,
    goal: 10,
    isCircular: true,
  };

  // Demo Card renders correctly
  const tree = renderer.create(
    <DemoCard
      name={mockCard.Name}
      number={mockCard.number}
      profilePicture={mockCard.profilePicture}
      demos={mockCard.demos}
      percentage={mockCard.percentage}
      madeCalls={mockCard.madeCalls}
      goal={mockCard.goal}
      isCircular={mockCard.isCircular}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();

  // Demo Card renders correctly without profile picture
  const tree2 = renderer.create(
    <DemoCard
      name={mockCard.Name}
      number={mockCard.number}
      profilePicture={null}
      demos={mockCard.demos}
      percentage={mockCard.percentage}
      madeCalls={mockCard.madeCalls}
      goal={mockCard.goal}
      isCircular={mockCard.isCircular}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
