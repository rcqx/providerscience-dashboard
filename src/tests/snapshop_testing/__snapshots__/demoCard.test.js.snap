// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex border-b border-slate-200 border-b-l border-b-r shadow-sm rounded-lg flex-1 py-2 xl:py-0 hover:scale-[103%] hover:border hover:border-slate-200 ease-out duration-300 cursor-pointer hover:bg-white"
>
  <div
    className="md:w-[15%] w-[25%]  flex justify-center items-center text-5xl font-bold text-slate-500 lg:text-5xl"
  >
    1
  </div>
  <div
    className="md:w-[20%] w-[25%] flex flex-col justify-center items-center"
  >
    <img
      alt="profile-pic"
      className="rounded-full border border-slate-300 h-[3em]"
      data-testid="profile-picture"
      src="aString"
    />
    <h1
      className="text-slate-700 font-bold text-sm"
    >
      John
    </h1>
  </div>
  <div
    className="flex-1 flex justify-center items-center"
  >
    <div
      className="relative"
    >
      <svg
        className="rc-progress-circle h-[70px] md:block hidden"
        role="presentation"
        style={Object {}}
        viewBox="-50 -50 100 100"
      >
        <circle
          className="rc-progress-circle-trail"
          cx={0}
          cy={0}
          r={46}
          stroke="#D9D9D9"
          strokeLinecap="round"
          strokeWidth={8}
          style={
            Object {
              "fillOpacity": 0,
              "stroke": "#D9D9D9",
              "strokeDasharray": "289.02652413026095px 289.02652413026095",
              "strokeDashoffset": 0,
              "transform": "rotate(-90deg)",
              "transformOrigin": "0 0",
              "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
            }
          }
        />
        <circle
          className="rc-progress-circle-path"
          cx={0}
          cy={0}
          opacity={1}
          r={46}
          strokeLinecap="round"
          strokeWidth={8}
          style={
            Object {
              "fillOpacity": 0,
              "stroke": "#4CBEEB",
              "strokeDasharray": "289.02652413026095px 289.02652413026095",
              "strokeDashoffset": 148.51326206513048,
              "transform": "rotate(-90deg)",
              "transformOrigin": "0 0",
              "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
            }
          }
        />
      </svg>
      <div>
        <h1
          className="absolute top-9 left-6 font-bold text-slate-700 hidden md:block"
        >
          50
          %
        </h1>
        <h1
          className=" md:flex flex-col items-center absolute bottom-8 marker:text-slate-600 hidden text-start w-full"
        >
          10
        </h1>
      </div>
    </div>
    <h1
      className="text-3xl font-bold md:hidden"
    >
      50
      %
    </h1>
  </div>
  <div
    className="flex-1 flex flex-col justify-center pr-1 md:pr-0"
  >
    <h1
      className="md:text-base lg:text-xl xl:text-base text-slate-600 font-bold line leading-tight hidden md:block lg:text-end lg:pr-[5%] xl:pr-[10%]"
    >
      5 Booked Demos
    </h1>
    <h1
      className="text-2xl font-bold md:hidden"
    >
      5
    </h1>
    <h2
      className="text-slate-700 text-xs md:text-sm lg:text-end lg:pr-[5%] xl:pr-[10%]"
    >
      1 Calls Made
    </h2>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="flex border-b border-slate-200 border-b-l border-b-r shadow-sm rounded-lg flex-1 py-2 xl:py-0 hover:scale-[103%] hover:border hover:border-slate-200 ease-out duration-300 cursor-pointer hover:bg-white"
>
  <div
    className="md:w-[15%] w-[25%]  flex justify-center items-center text-5xl font-bold text-slate-500 lg:text-5xl"
  >
    1
  </div>
  <div
    className="md:w-[20%] w-[25%] flex flex-col justify-center items-center"
  >
    <svg
      data-testid="default-profile-picture"
      fill="none"
      height={50}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
        fill="currentColor"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
    <h1
      className="text-slate-600 font-bold text-sm"
    >
      John
    </h1>
  </div>
  <div
    className="flex-1 flex justify-center items-center"
  >
    <div
      className="relative"
    >
      <svg
        className="rc-progress-circle h-[70px] md:block hidden"
        role="presentation"
        style={Object {}}
        viewBox="-50 -50 100 100"
      >
        <circle
          className="rc-progress-circle-trail"
          cx={0}
          cy={0}
          r={46}
          stroke="#D9D9D9"
          strokeLinecap="round"
          strokeWidth={8}
          style={
            Object {
              "fillOpacity": 0,
              "stroke": "#D9D9D9",
              "strokeDasharray": "289.02652413026095px 289.02652413026095",
              "strokeDashoffset": 0,
              "transform": "rotate(-90deg)",
              "transformOrigin": "0 0",
              "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
            }
          }
        />
        <circle
          className="rc-progress-circle-path"
          cx={0}
          cy={0}
          opacity={1}
          r={46}
          strokeLinecap="round"
          strokeWidth={8}
          style={
            Object {
              "fillOpacity": 0,
              "stroke": "#4CBEEB",
              "strokeDasharray": "289.02652413026095px 289.02652413026095",
              "strokeDashoffset": 148.51326206513048,
              "transform": "rotate(-90deg)",
              "transformOrigin": "0 0",
              "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
            }
          }
        />
      </svg>
      <div>
        <h1
          className="absolute top-9 left-6 font-bold text-slate-700 hidden md:block"
        >
          50
          %
        </h1>
        <h1
          className=" md:flex flex-col items-center absolute bottom-8 marker:text-slate-600 hidden text-start w-full"
        >
          10
        </h1>
      </div>
    </div>
    <h1
      className="text-3xl font-bold md:hidden"
    >
      50
      %
    </h1>
  </div>
  <div
    className="flex-1 flex flex-col justify-center pr-1 md:pr-0"
  >
    <h1
      className="md:text-base lg:text-xl xl:text-base text-slate-600 font-bold line leading-tight hidden md:block lg:text-end lg:pr-[5%] xl:pr-[10%]"
    >
      5 Booked Demos
    </h1>
    <h1
      className="text-2xl font-bold md:hidden"
    >
      5
    </h1>
    <h2
      className="text-slate-700 text-xs md:text-sm lg:text-end lg:pr-[5%] xl:pr-[10%]"
    >
      1 Calls Made
    </h2>
  </div>
</div>
`;
